doc/sphinx/make-your-own-compositor.md

```diff
@@ -0,0 +1,258 @@
+2. The reader has a C++ compiler with C++ 23 support.
+
+## A barebones Mir compositor
+This section will cover the needed dependencies and how to install them, the
+minimum code needed for a Mir compositor, how to build this code, and finally
+how to run your compositor.
+
+### Dependencies
+Before you start coding, you'll need to install `libmiral`, and
+`mir-graphics-drivers-desktop` which can be done on different distros as
+follows:
+
+Installing dependencies on Debian and its derivatives:
+```sh
+sudo apt install libmiral-dev mir-graphics-drivers-desktop
+```
+
+<details>
+<summary> Installing Dependencies on Other Distros </summary>
+
+Installing dependencies on Fedora 
+```sh
+sudo dnf install mir-devel libxkbcommon
+```
+Installing dependencies on Alpine 
+```sh
+sudo apk add mir-dev
+```
+</details>
+
+
+### Code
+The following code block is the bare minimum you need to run a Mir compositor:
+```cpp
+#include <miral/runner.h>
+#include <miral/minimal_window_manager.h>
+#include <miral/set_window_management_policy.h>
+
+using namespace miral;
+
+int main(int argc, char const* argv[])
+{
+    MirRunner runner{argc, argv};
+
+    return runner.run_with(
+        {
+            set_window_management_policy<MinimalWindowManager>()
+        });
+}
+```
+
+This program creates a floating window manager with basic window management
+capabilities such as controlling multiple windows, minimizing and maximizing,
+and handling mouse input. This is done with the help of MirAL (Mir Abstraction
+Layer) which gives you a high level interface to work with Mir.
+
+### Building
+To compile this simple program, you can use this `CMakeLists.txt` file:
+```cmake
+cmake_minimum_required(VERSION 3.5)
+
+project(demo-mir-compositor)
+
+set(CMAKE_CXX_STANDARD 23)
+
+include(FindPkgConfig)
+pkg_check_modules(MIRAL miral REQUIRED)
+pkg_check_modules(XKBCOMMON xkbcommon REQUIRED)
+
+add_executable(demo-mir-compositor main.cpp)
+
+target_include_directories(demo-mir-compositor PUBLIC SYSTEM ${MIRAL_INCLUDE_DIRS})
+target_link_libraries(     demo-mir-compositor               ${MIRAL_LDFLAGS})
+target_link_libraries(     demo-mir-compositor               ${XKBCOMMON_LIBRARIES})
+```
+
+Then to build:
+```sh
+cmake -B build
+cmake --build build
+```
+
+### Running
+To run, you can run nested in an X or Wayland session, or from a virtual
+terminal, just like the demo applications in [Explore What Mir Is Capable of
+Using Mir Demos](explore-mir-using-demos.md). For example, if we were to run
+inside an existing Wayland session, we'd run the following command:
+```sh
+WAYLAND_DISPLAY=wayland-99 ./build/demo-mir-compositor 
+```
+You'll see a window housing the compositor with a black void filling it. To
+fill this void with some content, you can run the following from another
+terminal:
+```sh
+WAYLAND_DISPLAY=wayland-99 bomber
+```
+You're free to replace `bomber` with any other Wayland compatible application.
+
+## Startup Applications
```

Comment:
I'm wondering now how many of these steps belong in "the" tutorial. There are many possibilities and we have to stop somewhere.

Maybe the tutorial ends here and each feature has a separate "HOWTO" starting from the above? Maybe there are a few "universal" features? Which?

Reply:
I think the tutorial should at end with a usable compositor, which would be the base + shortcuts

Reply:
"Usable" is quite a flexible concept: for example, Frame is usable without shortcuts.

I'm OK with shortcuts, or even shortcuts and startup. Just flagging we need to draw a line.

@mattkae where would you put the line

Reply:
I would draw the line at getting a simple window manager up and running. I think that the specifics of "How to do X in Mir" will each be a separate how-to that is linked at the bottom of the tutorial, e.g.:

```md
Now that you have your base compositor working, feel free to check out these guides on how to further develop your compositor:

- How to specify start up applications
- How to handle user input
- How to enable X11 applications
```

Reply:
Unresolving as the review comments suggest changes that haven't happened

Reply:
The later sections were split off into their own how-tos, anything else missing?